presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/RuleIndex.java

```diff
@@ -12,28 +12,29 @@
  * limitations under the License.
  */
 
-package com.facebook.presto.matching;
+package com.facebook.presto.sql.planner.iterative;
 
+import com.facebook.presto.matching.Pattern;
 import com.google.common.collect.ImmutableListMultimap;
 import com.google.common.collect.ListMultimap;
 import com.google.common.reflect.TypeToken;
 
 import java.util.Set;
 import java.util.stream.Stream;
 
-public class MatchingEngine<T extends Matchable>
+public class RuleIndex
 {
-    private final ListMultimap<Class, T> matchablesByClass;
+    private final ListMultimap<Class, Rule> rulesByRootType;
```

Comment:
there is another usage of this class where different type of rules is used.
See https://github.com/Teradata/presto/blob/release-0.179-t/presto-main/src/main/java/com/facebook/presto/cost/ComposableStatsCalculator.java.

I would prefer to make it possible to keep this usage there.